### ISSUE TITLE
typing.Union doc incoherence in case a class and its subclass are present

### ISSUE THREAD

**Issue by 5d296087-ec50-450a-a96a-e8dcaa06273f** [python/cpython#74804]
BPO | [30619](https://bugs.python.org/issue30619)
--- | :---
Nosy | @ilevkivskyi, @khyox, @Mariatta
PRs | <li>python/cpython#2326</li><li>python/cpython#2337</li><li>python/cpython#2338</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-06-23.02:18:36.636>
created_at = <Date 2017-06-10.00:38:10.682>
labels = ['3.7', 'docs']
title = 'typing.Union doc incoherence in case a class and its subclass are present'
updated_at = <Date 2017-06-23.02:18:36.630>
user = 'https://github.com/khyox'
```

bugs.python.org fields:
```python
activity = <Date 2017-06-23.02:18:36.630>
actor = 'Mariatta'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-06-23.02:18:36.636>
closer = 'Mariatta'
components = ['Documentation']
creation = <Date 2017-06-10.00:38:10.682>
creator = 'khyox'
dependencies = []
files = []
hgrepos = []
issue_num = 30619
keywords = []
message_count = 6.0
messages = ['295593', '296219', '296627', '296658', '296660', '296661']
nosy_count = 4.0
nosy_names = ['docs@python', 'levkivskyi', 'khyox', 'Mariatta']
pr_nums = ['2326', '2337', '2338']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30619'
versions = ['Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by 5d296087-ec50-450a-a96a-e8dcaa06273f** [comment]
In https://docs.python.org/3.6/library/typing.html#typing.Union, when giving details for Union and in the case when a class and its subclass are present, the doc states:

    When a class and its subclass are present, the former is skipped, e.g.:

        Union[int, object] == object

But it is the example (and not the text) which is consistent with PEP-483:

    Corollary: Union[..., object, ...] returns object

So, the proposed correction would be substitute 'former' for 'latter':

    When a class and its subclass are present, the latter is skipped, e.g.:

        Union[int, object] == object

**Comment by ilevkivskyi** [comment]
Yes, you are right. Could you please make a PR at https://github.com/python/cpython/pulls ?

**Comment by Mariatta** [comment]
New changeset 6580c19bbbe7bc9bc0884699afd69184f523b32e by Mariatta (khyox) in branch 'master':
bpo-30619: Clarify typing.Union documentation (GH-2326)
https://github.com/python/cpython/commit/6580c19bbbe7bc9bc0884699afd69184f523b32e

**Comment by Mariatta** [comment]
New changeset 8baf78b98f3acd76e51a4381542ff1612911b18f by Mariatta in branch '3.6':
[3.6] bpo-30619: Clarify typing.Union documentation (GH-2326) (GH-2337)
https://github.com/python/cpython/commit/8baf78b98f3acd76e51a4381542ff1612911b18f

**Comment by Mariatta** [comment]
New changeset ce1bd6ac7ffaf396157a9ceb55b281a3b196323f by Mariatta in branch '3.5':
[3.5] bpo-30619: Clarify typing.Union documentation (GH-2326) (GH-2338)
https://github.com/python/cpython/commit/ce1bd6ac7ffaf396157a9ceb55b281a3b196323f

**Comment by Mariatta** [comment]
Fixed and backported to 3.5 and 3.6.
Thanks!
